nautobot/ui/src/components/FiltersPanel.js

```diff
@@ -1,22 +1,41 @@
 import {
     Box,
+    Button,
     calc,
     CloseButton,
+    Divider,
     Flex,
+    FormContainer,
+    FormControl,
+    FormLabel,
     forwardRef,
     getCssVar,
     Heading,
+    Input,
     NtcThumbnailIcon,
+    ReactSelect,
+    Tag,
+    TagCloseButton,
+    TagLabel,
 } from "@nautobot/nautobot-ui";
 import React, {
+    cloneElement,
     createContext,
+    Fragment,
+    isValidElement,
     useCallback,
     useContext,
     useEffect,
     useMemo,
     useRef,
     useState,
 } from "react";
+import { useSearchParams } from "react-router-dom";
+
+import { useLazyGetFieldLookupChoicesQuery } from "@utils/api";
+
+export const FILER_RESET_QUERY_PARAMS = ["offset"];
+export const NON_FILTER_QUERY_PARAMS = ["limit", "offset", "sort"];
```

Comment:
Is it a right call to list query params, which are not filters, and assume all others are filters? If so, are these 3 correct?

Reply:
This _might_ be fine for now, but I think a better way to do this is to emit the FilterSet as a JSON schema along with the model schema, something like

```json
{
    "name":
        "__e": "Exact",
        "__ic": "Contains (Case-Insensitive)",
   #...
}
```

And you can check query params against the dictionary.

OR

In the API response of a filtered query, return a `_meta` property with interpreted filtered params:

```json
{
    "_meta": {
        "filters": {
            "name__ic": "ams"
        }
    },
    "count": 11,
    "next": "https://next.demo.nautobot.com/api/dcim/devices/?limit=5&name__ic=ams&offset=5",
    "previous": null,
    "results": [ #... ]
}
```

Reply:
That makes sense, I'll update the list. Though, since they are already hardcoded in backend, I do not see any reason not to do that in the UI. That is, unless they can be emitted via HTTP endpoint which the UI can be hoooked up to

Reply:
Anyway, if these names are static, perhaps we do not even have complicate things with additional API endpoints, and instead move them to a shared JSON or YAML file which both API and UI can import.